### ISSUE TITLE
GPT3 训练不支持使用PPL作为evaluate metrics

### ISSUE THREAD

**Issue by David-Lee-1990** [modelscope/modelscope#158]
您好，我们发现目前的代码中，Distributed GPT3不支持使用PPL作为evaluate metric，因为每次都会走generate函数，而generate函数不输出logits向量。

目前可以自己继承相关类重写GPT3类，但是这样还是不太方便。希望后面能够考虑更多的可能性，给予比较好的参数设定方式。

**Comment by Firmament-cyou** [comment]
> 您好，我们发现目前的代码中，Distributed GPT3不支持使用PPL作为evaluate metric，因为每次都会走generate函数，而generate函数不输出logits向量。
> 
> 目前可以自己继承相关类重写GPT3类，但是这样还是不太方便。希望后面能够考虑更多的可能性，给予比较好的参数设定方式。

您好，GPT3 训练过程中会根据输入数据是否包含 tgt_txt 选择使用 forward_eval 还是 generate_eval，可以参考 modelscope/trainers/nlp/gpt3_trainer.py 中的代码
